h2/connection.py

```diff
@@ -1387,7 +1387,9 @@ def _receive_headers_frame(self, frame):
             # compatibility, catch all of them.
             raise ProtocolError("Error decoding header block: %s" % e)
 
-        headers = validate_headers(headers)
+        headers = validate_headers(headers,
+                                   client_side=self.client_side,
+                                   is_trailer=False)
```

Comment:
So I appreciate the plan here, but I definitely think we want to swap this to a namedtuple before we continue. We'll likely have to add more filter criteria here, and so avoiding the proliferation of kwargs would be ideal.